test/integration/smimetrics/smi-metrics_test.go

```diff
@@ -0,0 +1,168 @@
+package smimetrics
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/testutil"
+	"github.com/servicemeshinterface/smi-sdk-go/pkg/apis/metrics/v1alpha1"
+)
+
+var TestHelper *testutil.TestHelper
+
+type testCase struct {
+	name string
+	kind string
+	// edges > 0 denotes that its a edges query, otherwise a resource query
+	edges int
+}
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(testutil.Run(m, TestHelper))
+}
+
+func TestSMIMetrics(t *testing.T) {
+
+	testNamespace := TestHelper.GetTestNamespace("smi-metrics-test")
+	err := TestHelper.CreateDataPlaneNamespaceIfNotExists(testNamespace, nil)
+	if err != nil {
+		testutil.AnnotatedFatalf(t, "failed to create %s namespace: %s", testNamespace, err)
+	}
+
+	// Add the SMI Helm repo
+	args := []string{
+		"repo",
+		"add",
+		"smi",
+		"https://servicemeshinterface.github.io/smi-metrics/",
+	}
+
+	if stdout, stderr, err := TestHelper.HelmRun(args...); err != nil {
+		testutil.AnnotatedFatalf(t, "'helm repo add' command failed\n%s\n%s\n%v", stdout, stderr, err)
```

Comment:
I love that we have this integration test but I am a bit worried about issues with the SMI chart taking down our CI.  It would be nice to have some escape hatch so that if the SMI chart is having issues we can bypass this test.  Perhaps if the Helm installation of the SMI chart fails we should log an annotated warning and abort this integration test.

Do you think that would be visible enough?  Where are the annotated warning surfaced? @alpeb

Reply:
The warning annotation is shown in the CI run's main page (as soon as you click on a run in the Actions tab). Another option is to download the chart (as a tar.gz file) and add it as a test artifact.

Reply:
Updated the PR to have the chart in the file system. So that we don't have to an external dependency in CI